File: 1695-maximum-erasure-value/1695-maximum-erasure-value.java
class Solution {
    public int maximumUniqueSubarray(int[] nums) {
        HashSet<Integer> hashSet=new HashSet<>();
        int ans=0;
        int s=0;
        int l=0;
        for(int i: nums){
            while(hashSet.contains(i)){
                s-=nums[l];
                hashSet.remove(nums[l]);
                l++;
            }
            hashSet.add(i);
            s+=i;
            ans=Math.max(ans,s);
        }
        return ans;
    }
}

File: 1695-maximum-erasure-value/1695-maximum-erasure-value.py
class Solution:
    def maximumUniqueSubarray(self, nums: List[int]) -> int:
        hashSet=set()
        ans=s=l=0
        for i in nums:
            while i in hashSet:
                s-=nums[l]
                hashSet.remove(nums[l])
                l+=1
            hashSet.add(i)
            s+=i
            ans=max(ans,s)
        return ans

        